### ISSUE TITLE
Fails to exit process when used via husky

### ISSUE THREAD

**Issue by k2snowman69** [k2snowman69/exec-if-exists#415]
When the husky config looks like this:

```
module.exports = {
  hooks: {
    'commit-msg': [
      'echo npm',
      'exec-if-exists npm -v',
      'echo does not exist',
      'exec-if-exists asdfasdfasdfadsf -v',
      'echo commitlint',
      'exec-if-exists commitlint -E HUSKY_GIT_PARAMS',
    ].join(' && '),
    'pre-commit': 'lint-staged',
    'pre-push': 'npm test',
  },
};
```

The output looks like this:

```
husky > commit-msg (node v12.18.4)
npm
6.14.6
_stream_readable.js:681
    dest.end();
         ^

TypeError: dest.end is not a function
    at Socket.onend (_stream_readable.js:681:10)
    at Object.onceWrapper (events.js:421:28)
    at Socket.emit (events.js:327:22)
    at endReadableNT (_stream_readable.js:1220:12)
    at processTicksAndRejections (internal/process/task_queues.js:84:21)
husky > commit-msg hook failed (add --no-verify to bypass)
```

So looks like something is wrong with exiting the first process or maybe piping?

**Comment by k2snowman69** [comment]
I found a similar thing in Husky's documentation that was happening with Cypress
https://github.com/cypress-io/cypress/pull/5045/files

**Comment by k2snowman69** [comment]
Fixed with 3.0.2
